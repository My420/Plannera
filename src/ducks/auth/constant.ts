import { fb as firebase } from '../../firebase/config';

export const moduleName = 'auth';
export const SIGN_UP_REQUEST = 'Plannera/auth/SIGN_UP_REQUEST';
export const SIGN_UP_SUCCESS = 'Plannera/auth/SIGN_UP_SUCCESS';
export const SIGN_UP_ERROR = 'Plannera/auth/SIGN_UP_ERROR';
export const SIGN_IN_REQUEST = 'Plannera/auth/SIGN_IN_REQUEST';
export const SIGN_IN_SUCCESS = 'Plannera/auth/SIGN_IN_SUCCESS';
export const SIGN_IN_ERROR = 'Plannera/auth/SIGN_IN_ERROR';
export const SIGN_OUT_REQUEST = 'Plannera/auth/SIGN_OUT_REQUEST';
export const SIGN_OUT_SUCCESS = 'Plannera/auth/SIGN_OUT_SUCCESS';
export const SIGN_OUT_ERROR = 'Plannera/auth/SIGN_OUT_ERROR';
export const AUTH_CLEAR_ERROR = 'Plannera/auth/AUTH_CLEAR_ERROR';

export const auth = firebase.auth();
