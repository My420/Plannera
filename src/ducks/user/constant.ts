export const moduleName = 'user';
export const CREATE_USER_REQUEST = 'Plannera/user/CREATE_USER_REQUEST';
export const CREATE_USER_SUCCESS = 'Plannera/user/CREATE_USER_SUCCESS';
export const CREATE_USER_ERROR = 'Plannera/user/CREATE_USER_ERROR';

export const LOAD_USER_REQUEST = 'Plannera/user/LOAD_USER_REQUEST';
export const LOAD_USER_SUCCESS = 'Plannera/user/LOAD_USER_SUCCESS';
export const LOAD_USER_ERROR = 'Plannera/user/LOAD_USER_ERROR';

export const CHANGE_USER_DATA_REQUEST = 'Plannera/user/CHANGE_USER_DATA_REQUEST';
export const CHANGE_USER_DATA_SUCCESS = 'Plannera/user/CHANGE_USER_DATA_SUCCESS';
export const CHANGE_USER_DATA_ERROR = 'Plannera/user/CHANGE_USER_DATA_ERROR';

export const CLEAR_USER_STATE = 'Plannera/user/CLEAR_USER_STATE';
