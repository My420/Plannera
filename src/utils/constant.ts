export const APP_NAME = 'Plannera';

export const EMAIL = 'EMAIL';
export const PASSWORD = 'PASSWORD';
export const CONFIRM = 'CONFIRM';
export const FIRST_NAME = 'FIRST_NAME';
export const LAST_NAME = 'LAST_NAME';

export const MAIN_PAGE = '/';
export const SIGN_UP_PAGE = '/signup';
export const SIGN_IN_PAGE = '/signin';

export const DB_USER_CATALOG = 'users';
