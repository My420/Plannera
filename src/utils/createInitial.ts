const createInitial = (firstName: string, lastName: string): string => {
  const initial = (firstName[0] + lastName[0]).toUpperCase();
  return initial;
};

export default createInitial;
