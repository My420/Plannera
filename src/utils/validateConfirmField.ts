import { IFieldValidator } from '../types/interfaces';

export interface IValidateConfirmField {
  (value: string, password: string): IFieldValidator;
}

export const ERROR_NOT_MATCH: IFieldValidator = {
  isValid: false,
  error: 'does not match password field',
};
export const VALID_CONFIRM: IFieldValidator = { isValid: true, error: '' };

// eslint-disable-next-line max-len
const validateConfirmField: IValidateConfirmField = (value, password) => (value === password ? VALID_CONFIRM : ERROR_NOT_MATCH);

export default validateConfirmField;
