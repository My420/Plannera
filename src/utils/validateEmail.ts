import { IFieldValidator } from '../types/interfaces';

// eslint-disable-next-line no-useless-escape
const tester = /^[-!#$%&'*+\/0-9=?A-Z^_a-z`{|}~](\.?[-!#$%&'*+\/0-9=?A-Z^_a-z`{|}~])*@[a-zA-Z0-9](-*\.?[a-zA-Z0-9])*\.[a-zA-Z](-?[a-zA-Z0-9])+$/;
const ERROR = {
  EMPTY: 'field cannot be empty.',
  TOLONG: 'email address is too long',
  NOTMATCH: 'invalid email address',
};

const validateEmail = (email: string): IFieldValidator => {
  if (!email) return { isValid: false, error: ERROR.EMPTY };

  if (email.length > 254) return { isValid: false, error: ERROR.TOLONG };

  const valid = tester.test(email);
  if (!valid) return { isValid: false, error: ERROR.NOTMATCH };

  // Further checking of some things regex can't handle
  const parts = email.split('@');
  if (parts[0].length > 64) return { isValid: false, error: ERROR.NOTMATCH };

  const domainParts = parts[1].split('.');
  if (domainParts.some((part) => part.length > 63)) {
    return { isValid: false, error: ERROR.NOTMATCH };
  }

  return { isValid: true, error: '' };
};

export default validateEmail;
