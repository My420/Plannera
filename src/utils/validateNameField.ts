import { IFieldValidator, IErrorConstant } from '../types/signUpForm';

export interface IValidateNameField {
  (value: string): IFieldValidator;
}

export const ERROR: IErrorConstant = {
  NOT_VALID_SYMBOL: { isValid: false, error: 'only letters and numbers' },
  TO_SHORT: { isValid: false, error: 'must be longer than 2 characters' },
  TO_LONG: { isValid: false, error: 'must be less than 20 characters long' },
};

export const VALID_NAME_FIELD = { isValid: true, error: '' };

const validateNameField: IValidateNameField = (value) => {
  const tester = /[^A-Za-z0-9]+/;
  const valid = tester.test(value);
  if (valid) {
    return ERROR.NOT_VALID_SYMBOL;
  }
  const { length } = value;

  if (length < 2) {
    return ERROR.TO_SHORT;
  }

  if (length > 20) {
    return ERROR.TO_LONG;
  }

  return VALID_NAME_FIELD;
};

export default validateNameField;
