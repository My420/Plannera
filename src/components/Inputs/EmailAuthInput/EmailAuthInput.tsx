import React from 'react';
import { EMAIL } from '../../../utils/constant';
import AuthInput from '../AuthInput';

export interface EmailAuthInputProps {
  extensionContainerClass?: string;
  extensionInputClass?: string;
  value: string;
  isValid: boolean;
  error: string;
  onInputChange: (name: string, value: string) => void;
}

const EmailAuthInput: React.FC<EmailAuthInputProps> = (props) => (
  <AuthInput
    id="email"
    name={`${EMAIL}`}
    type="email"
    placeholder="Email"
    label="Email"
    {...props}
  />
);

EmailAuthInput.defaultProps = {
  extensionContainerClass: '',
  extensionInputClass: '',
};

export default EmailAuthInput;
