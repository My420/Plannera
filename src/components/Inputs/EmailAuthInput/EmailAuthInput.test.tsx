import React from 'react';
import { render, cleanup, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import EmailAuthInput, { EmailAuthInputProps } from './EmailAuthInput';
import { EMAIL } from '../../../utils/constant';

describe('test EmailAuthInput component', () => {
  const mockFn = jest.fn();
  const props: EmailAuthInputProps = {
    value: '',
    isValid: false,
    error: '',
    onInputChange: mockFn,
  };

  const expected = {
    textLabel: 'Email',
    placeholder: 'Email',
    type: 'email',
    name: EMAIL,
  };

  afterEach(cleanup);

  test('should match snapshot', () => {
    const { asFragment } = render(<EmailAuthInput {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('should render label that associate with input', () => {
    render(<EmailAuthInput {...props} />);
    const elem = screen.getByLabelText(expected.textLabel);
    expect(elem.tagName).toBe('INPUT');
  });

  test("should render tag span with label text and 'visually-hidden' class", () => {
    render(<EmailAuthInput {...props} />);
    const elem = screen.getByText(expected.textLabel);
    expect(elem).toHaveClass('visually-hidden');
  });

  test('should render tag input with correct attr', () => {
    render(<EmailAuthInput {...props} />);
    const elem = screen.getByPlaceholderText(expected.placeholder);
    expect(elem.tagName).toBe('INPUT');
    expect(elem).toHaveValue(props.value);
    expect(elem).toHaveAttribute('type', expected.type);
  });

  test('should render correct tag span with error if props.isValid = false', () => {
    render(<EmailAuthInput {...props} />);
    const elem = screen.getByTestId('error');
    expect(elem).toHaveClass('error');
    expect(elem).toHaveTextContent(props.error);
  });

  test('should render correct tag span with error if props.isValid = true', () => {
    const errorText = 'error error';
    const newProps = { ...props, isValid: true, error: errorText };
    render(<EmailAuthInput {...newProps} />);
    const elem = screen.getByText(errorText);
    expect(elem).toHaveClass('valid');
  });

  test('when user enters text should call props.onInputChange function', async () => {
    render(<EmailAuthInput {...props} />);
    const elem = screen.getByPlaceholderText(expected.placeholder);
    const text = 'test user input';
    await userEvent.type(elem, text, { allAtOnce: true });
    expect(mockFn).toHaveBeenCalledWith(expected.name, text);
  });
});
