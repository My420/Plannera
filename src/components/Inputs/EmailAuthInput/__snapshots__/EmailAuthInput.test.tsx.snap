// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test EmailAuthInput component should match snapshot 1`] = `
<DocumentFragment>
  <label
    class="label "
    for="email"
  >
    <span
      class="visually-hidden"
    >
      Email
    </span>
    <span
      class="error"
      data-testid="error"
    />
    <input
      class="input "
      id="email"
      name="EMAIL"
      placeholder="Email"
      type="email"
      value=""
    />
  </label>
</DocumentFragment>
`;
